std/object/client_trust.go

```diff
@@ -0,0 +1,83 @@
+package object
+
+import (
+	"fmt"
+
+	enc "github.com/named-data/ndnd/std/encoding"
+	"github.com/named-data/ndnd/std/ndn"
+	sec "github.com/named-data/ndnd/std/security"
+	"github.com/named-data/ndnd/std/security/signer"
+)
+
+// SuggestSigner returns the signer for a given name
+// nil is returned if no signer is found
+func (c *Client) SuggestSigner(name enc.Name) ndn.Signer {
+	if c.trust == nil {
+		return signer.NewSha256Signer()
+	}
+	name = rmSegVer(name)
+	return c.trust.Suggest(name)
+}
+
+// Validate a data packet using the client configuration
+func (c *Client) Validate(data ndn.Data, sigCov enc.Wire, callback func(bool, error)) {
+	c.ValidateExt(ndn.ValidateExtArgs{
+		Data:       data,
+		SigCovered: sigCov,
+		Callback:   callback,
+	})
+}
+
+// ValidateExt is an advanced API for validating data packets
+func (c *Client) ValidateExt(args ndn.ValidateExtArgs) {
+	if c.trust == nil {
+		args.Callback(true, nil)
+		return
+	}
+
+	// Pop off the version and segment components
+	overrideName := rmSegVer(args.Data.Name())
+	if len(args.OverrideName) > 0 {
+		overrideName = args.OverrideName
+	}
+
+	// Add to queue of validation
+	select {
+	case c.validatepipe <- sec.TrustConfigValidateArgs{
+		Data:         args.Data,
+		DataSigCov:   args.SigCovered,
+		Callback:     args.Callback,
+		OverrideName: overrideName,
+		Fetch: func(name enc.Name, config *ndn.InterestConfig, callback ndn.ExpressCallbackFunc) {
+			// Pass through extra options
+			if args.CertNextHop != nil {
+				config.NextHopId = args.CertNextHop
+			}
+
+			// Express the interest with reliability
+			c.ExpressR(ndn.ExpressRArgs{
+				Name:     name,
+				Config:   config,
+				Retries:  3,
+				Callback: callback,
+			})
+		},
+	}:
+		// Queued successfully
+	default:
+		args.Callback(false, fmt.Errorf("validation queue full"))
+	}
+}
+
+// rmSegVer removes the segment and version components from a name
+func rmSegVer(name enc.Name) enc.Name {
```

Comment:
Let's call it `removeSegVer`. `rm` is too nerdy in a function name.

Reply:
Done